WallsAndHoles/mapcell.h

```diff
@@ -1,37 +1,50 @@
 #ifndef MAPCELL_H
 #define MAPCELL_H
 
+#include "mapcellgraphicsitem.h"
 #include "tile.h"
 
 #include <QObject>
 #include <QGraphicsScene>
-#include <QGraphicsItem>
-#include <QGraphicsRectItem>
 #include <QGraphicsSceneHoverEvent>
 
 class MapCell : public QObject
 {
     Q_OBJECT
 
 public:
+
     MapCell(QGraphicsScene *scene, int x, int y, const Tile &tile, QObject *parent = nullptr);
     ~MapCell();
 
     void setHighlightBrush(const QBrush &color) { mHighlight->setBrush(color); }
 
+    /**
+     * @brief setGraphics changes the view of the class
+     * @param flag:
+     * (0 - default view)
+     * (1 - height map view)
+     */
+    void setGraphics(int viewFlag);
+
 public slots:
     void tileChanged();
 
 private:
+
     QGraphicsScene *mScene;
     int mX, mY;
 
     //the tile this cell is displaying.
     const Tile &mTile;
 
-    QGraphicsRectItem *mGraphics;
-    QGraphicsRectItem *mHighlight;
-    QGraphicsRectItem *mBackground;
+    enum views { defaultView, heightMapView };
+    int mViewFlag;
+
+    MapCellGraphicsItem *mBackground;
```

Comment:
Any reason to have mBackground or mHighlight be MapCellGraphicsItems instead of QRectGraphicsItems?